nuxeo-core/nuxeo-core-storage-dbs/src/main/java/org/nuxeo/ecm/core/storage/dbs/DBSSession.java

```diff
@@ -887,17 +894,21 @@ protected void remove(String rootId) {
             }
         };
         collector.accept(rootState); // add the root node too
-        try (Stream<State> states = transaction.getDescendants(rootId, KEYS_RETENTION_ACTIVE_AND_PROXIES, 0)) {
+        try (Stream<State> states = transaction.getDescendants(rootId, KEYS_UNDELETABLE_AND_PROXIES, 0)) {
             states.forEach(collector);
         }
 
-        // if a subdocument is under active retention, removal fails
-        if (!retentionActiveIds.isEmpty()) {
-            if (retentionActiveIds.contains(rootId)) {
-                throw new DocumentExistsException("Cannot remove " + rootId + ", it is under active retention");
-            } else {
-                throw new DocumentExistsException("Cannot remove " + rootId + ", subdocument "
-                        + retentionActiveIds.iterator().next() + " is under active retention");
+        // if a subdocument is under retention / hold, removal fails
+        if (!undeletableIds.isEmpty()) {
+            // in tests we may want to delete everything
+            boolean allowDeleteUndeletable = Framework.isBooleanPropertyTrue("allowDeleteUndeletableDocuments");
```

Comment:
Isn't something we want to have as a feature? If so, maybe worth renaming it to something like: `nuxeo.retention.disable.undeletable`, wdyt?

Reply:
I prefer to keep it internal (tests) for now, and we'll rename it if/when it's needed externally.